pkg/activator/handler/metric_handler.go

```diff
@@ -49,17 +49,25 @@ func (h *MetricHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	start := time.Now()
 
 	rr := pkghttp.NewResponseRecorder(w, http.StatusOK)
-	defer func() {
+	defer func(isTLS bool) {
 		err := recover()
 		latency := time.Since(start)
 		if err != nil {
 			reporterCtx := metrics.AugmentWithResponse(reporterCtx, http.StatusInternalServerError)
+			if isTLS {
+				pkgmetrics.Record(reporterCtx, tlsConnectionCountM.M(1))
+			}
```

Comment:
Rather than reporting a second metric, how about adding a tag to the existing metric, ala:

```golang
securityMode := // get security mode from context
reporterCtx, _ := tag.New(reporterCtx, tag.Upsert(SecurityMode, securityMode))
```

That will also provide a tag which you can use to check the mutual-TLS mode when it's implemented, and doesn't need to expose an extra metric.

Reply:
ah cool, I didn't know about tags. That makes sense, this will be easier to expand when the rest of the trust levels are implemented.

I'll make a first attempt at this where I pass the trustmode into the MetricHandler creation, instead of pulling it from Context. 

As far as I understand, the activator doesn't have a watcher for the config-network configmap, so I don't think you can pull it from the context. And I'm less confident about changing that right now.

Reply:
ah, now it does have that ability, so I can probably pull from context now: https://github.com/knative/serving/issues/13694